spec/factorial_spec.rb

```diff
@@ -0,0 +1,25 @@
+require './solver'
+
+describe Solver do
+  context 'When testing the Solver class' do
+    solver = Solver.new
+
+    it 'the method factorial should return the factorial of given integer' do
+      value = 5
+      result = solver.factorial(value)
+      expect(result).to eq 120
+    end
+
+    it 'the method factorial should return the factorial of given integer' do
+      value = 4
+      result = solver.factorial(value)
+      expect(result).to eq 24
+    end
+
+    it 'the method factorial should return the factorial of given integer' do
+      value = 6
+      result = solver.factorial(value)
+      expect(result).to eq 720
+    end
+  end
+end
```

Comment:
- [ ] Since it is a requirement for the this project and best practice, testing edge cases, it will be nice to add a test case which handles a `negative value and raises an exception as an expectation`
- [ ] It will also be nice to add tests for special cases of 0 and 1.